### PR TITLE
repo-updater: Add configurable rate limiter for GitHub

### PR DESCRIPTION
**Pull request by ryanslade**
Adds rate limiter config for GitHub

Stacked on: https://github.com/sourcegraph/sourcegraph/pull/9653

Part of: https://github.com/sourcegraph/sourcegraph/issues/8546

### CODE REVIEW

File: cmd/repo-updater/repos/types.go [hunk 1]
```diff
@@ -1003,6 +1003,16 @@ func (r *RateLimiterRegistry) updateRateLimiter(svc *ExternalService) error {
...
+	case *schema.GitHubConnection:
+		// 5000 by hour is the default enforced by GitHub on their end
+		limit = rate.Limit(5000.0 / 3600.0)
+		if c != nil && c.RateLimit != nil {
+			if c.RateLimit.Enabled {
+				limit = rate.Limit(c.RateLimit.RequestsPerHour / 3600)
+			} else {
+				limit = rate.Inf
+			}
+		}
```
Comment (unknwon) [inline cmd/repo-updater/repos/types.go:1015]: As I commented on #9653, the else branch will make the default assignment ineffective (always overwrite).

**Comment by ryanslade** [comment]
Sorry about the review spam. I rebased and that seems to have triggered a mass review request :)

**Comment by unknwon** [comment]
I was thinking a better workflow because this is going to happen every time a merge conflict occurs (which is a very bad behavior from GitHub). We probably could make stacked PRs as draft and request reviews directly, and once the base PR is merged, we check if there is any conflict, if no, then mark as ready. I know @rvantonder does a lot stacked PRs, maybe could share some tricks? :D

**Comment by rvantonder** [comment]
@unknwon 

> maybe could share some tricks? :D

Eh, it's kinda error prone with GH. There's more than one way to do this, but I basically do, for each part of the stack, after review:
- make sure a previous PR on the stack is merged into master
- git pull master locally
- take note of the commit for "this" PR (which I will cherry-pick later).
- `git reset --hard` to the commit before the stack

**Comment by unknwon** [comment]
Thanks for sharing!

> I don't keep more than one PR stack going at once, this is a sure way to forget a step and mess up something. Also, the review spam is GH's fault because of the rebase changes that can trigger when the base is changed 🤷

I totally agree, GitHub should have done this better.

**Pull request merged**